suite.go

```diff
@@ -212,9 +212,9 @@ func (s *suite) runBeforeStepHooks(ctx context.Context, step *Step, err error) (
 	return ctx, err
 }
 
-func (s *suite) runAfterStepHooks(ctx context.Context, step *Step, status StepResultStatus, err error) (context.Context, error) {
-	for _, f := range s.afterStepHandlers {
-		hctx, herr := f(ctx, step, status, err)
+func (suite *testSuite) runAfterStepHooks(ctx context.Context, step *Step, status StepResultStatus, err error) (context.Context, error) {
+	for _, handler := range suite.afterStepHandlers {
+		hctx, herr := handler(ctx, step, status, err)
```

Comment:
Hmm. Scrolling down to this example also seems to muddy things a little further. What also adds more to the confusion is that what are called handlers here are actually hooks, so there's an odd naming swap. I haven't looked deep enough to see if there's a particular reason why though.

When trying to understand just what calling `handler` did, it took a little more digging than would probably be good for others working on the code. I wonder if there's a slightly different way of organizing things where we could change Hook to be a type/struct or even an interface so that we could do something like, `Hook.run()`? That seems to read a little better anyhow.

Reply:
> > I guess this may not be idiomatic Go? I'm ready to hear feedback on that!
> 
> You're correct here. Since method receivers are generally very important things in their methods, a one letter assignment is common. The other place where single letters are generally suggested are within things like loops/ranges. [This post](https://dave.cheney.net/practical-go/presentations/qcon-china.html#_use_a_consistent_naming_style) is a **fantastic** read. I linked to a particular section on naming and what's most pertinent to this topic is the first `tip` comment within it.

Thanks! These terse names make more sense now. I definitely want the code to be idiomatic, so I'll change these back.

Reply:
Ideally I would like to see an abstraction wrapped around both the step definitions and the hooks that lets us treat all of them as "test steps" which we can just compile into one big list and execute in turn. But I think that's for another PR!